transitions/core.py

```diff
@@ -284,13 +285,27 @@ def _trigger(self, model, *args, **kwargs):
                 return False
             else:
                 raise MachineError(msg)
-        event = EventData(state, self, self.machine, model,
-                          args=args, kwargs=kwargs)
-        for t in self.transitions[state.name]:
-            event.transition = t
-            if t.execute(event):
-                return True
-        return False
+        event_data = EventData(state, self, self.machine, model, args=args, kwargs=kwargs)
+
+        for func in self.machine.prepare_event:
+            self.machine._callback(func, event_data)
+            logger.debug("Executed machine preparation callback '%s' before conditions." % func)
+
+        result = False
+        try:
+            for t in self.transitions[state.name]:
+                event_data.transition = t
+                if t.execute(event_data):
+                    result = True
```

Comment:
Can we store the `result` on the `event_data.result` so `finalize_event` knows that?
Do we have time before 4.4 release?